src/amo/components/CollectionManager/index.js

```diff
@@ -104,13 +108,17 @@ export class CollectionManagerBase extends React.Component<Props, State> {
     }
 
     if (hasAddonBeenAddedNew && hasAddonBeenAddedNew !== hasAddonBeenAdded) {
-      this.props.setTimeout(
+      this.timeout = this.props.setTimeout(
         this.resetMessageStatus,
         MESSAGE_RESET_TIME
       );
     }
   }
 
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
```

Comment:
What happens if you call `clearTimeout` with `this.timeout === undefined` (or `null`)? I guess we could play safe and write:

```js
if (this.timeout) {
  clearTimeout(this.timeout);
}
```

Reply:
I think we should also cover this line with a test case.

We could inject `clearTimeout` like `setTimeout` and use `unmount()` from Enzyme: http://airbnb.io/enzyme/docs/api/ShallowWrapper/unmount.html.

Reply:
yes! let me look into this more : )

Reply:
@willdurand, I have a shell set up for this but i may need a little bit of help. I want to try to dig into this a bit before first tho : ). I'll let you know...